pkg/cloudprovider/provider/digitalocean/provider.go

```diff
@@ -268,19 +266,20 @@ func (p *provider) Create(machine *v1alpha1.Machine, userdata string) (instance.
 	ctx := context.TODO()
 	client := getClient(c.Token)
 
-	// TODO(p0lyn0mial): an ssh-key should be created only if the CoreOS was requested
-	// TODO(p0lyn0mial): an ssh-key must be temporal - deleted after an instance has been created.
-	if p.privateKey == nil {
-		var err error
-		p.privateKey, err = machinessh.NewPrivateKey("machine-controller")
-		if err != nil {
-			return nil, err
-		}
+	tmpSSHKeyPair, err := machinessh.NewPrivateKey("machine-controller-do-tmp-key")
+	if err != nil {
+		return nil, err
 	}
-	fingerprint, err := ensureSSHKeysExist(ctx, client.Keys, p.privateKey)
+	fingerprint, err := ensureSSHKeysExist(ctx, client.Keys, tmpSSHKeyPair)
 	if err != nil {
-		return nil, fmt.Errorf("failed ensure that the ssh key '%s' exists: %v", p.privateKey.Name(), err)
+		return nil, fmt.Errorf("failed ensure that the ssh key '%s' exists: %v", tmpSSHKeyPair.Name(), err)
 	}
+	defer func() {
+		_, err := client.Keys.DeleteByFingerprint(ctx, fingerprint)
+		if err != nil {
```

Comment:
not sure what else we can do in case of an error. I was thinking add retry logic but I am pretty sure it is already implemented by `client.Keys` library

Reply:
can we set TTL on an ssh-key :D ?

Reply:
Keep it simple, simply logging the error is fine.
In the rare case this really fails, it does not hurt.